Services/Authentication/classes/class.ilAuthPlugin.php

```diff
@@ -8,7 +8,7 @@
  *
  * @author Stefan Meyer <smeyer.ilias@gmx.de>
  */
-abstract class ilAuthPlugin extends ilPlugin
+abstract class ilAuthPlugin extends ilPlugin implements ilAuthDefinition
```

Comment:
if I create a new plugin, i need to know, which method needs to implement, so the abstract plugin class needs to implement this interface and not the plugin class itself
better would be to seperate the definition to a seperate class and not the plugin class itself
no, it's not incompatible to other plugins, because they implements the interface and/or these methods itself and PHP allows to implement an interface in a parent and a child class